nixos/tests/tpm-ek/default.nix

```diff
@@ -0,0 +1,109 @@
+import ../make-test-python.nix (
+  { lib, pkgs, ... }:
+
+  let
+    inherit (pkgs) writeText tpm2-tools openssl;
+    ek_config = writeText "ek-sign.cnf" ''
+      [ tpm_policy ]
+      basicConstraints = CA:FALSE
+
+      keyUsage = keyEncipherment
+      certificatePolicies = 2.23.133.2.1
+      extendedKeyUsage = 2.23.133.8.1
+
+      subjectAltName = ASN1:SEQUENCE:dirname_tpm
+
+      [ dirname_tpm ]
+      seq = EXPLICIT:4,SEQUENCE:dirname_tpm_seq
+
+      [ dirname_tpm_seq ]
+      set = SET:dirname_tpm_set
+
+      [ dirname_tpm_set ]
+      seq.1 = SEQUENCE:dirname_tpm_seq_manufacturer
+      seq.2 = SEQUENCE:dirname_tpm_seq_model
+      seq.3 = SEQUENCE:dirname_tpm_seq_version
+
+      # We're going to mock up an STM TPM here
+      [dirname_tpm_seq_manufacturer]
+      oid = OID:2.23.133.2.1
+      str = UTF8:"id:53544D20"
+
+      [dirname_tpm_seq_model]
+      oid = OID:2.23.133.2.2
+      str = UTF8:"ST33HTPHAHD4
+
+      [dirname_tpm_seq_version]
+      oid = OID:2.23.133.2.3
+      str = UTF8:"id:00010101"
+    '';
+  in
+  {
+    name = "tpm-ek";
+
+    meta = {
+      maintainers = with lib.maintainers; [ baloo ];
+    };
+
+    nodes.machine =
+      { pkgs, ... }:
+      {
+        environment.systemPackages = [
+          openssl
+          tpm2-tools
+        ];
+
+        security.tpm2 = {
+          enable = true;
+          tctiEnvironment.enable = true;
+        };
+
+        virtualisation.tpm = {
+          enable = true;
+          provisioning = ''
+            export PATH=${
```

Comment:
Should we add tpm2-tools to PATH just prior to execution of the provisioning hook? I foresee just about everyone needing the tpm2_* utilities in here (and the example in the option module doesn't have a prefixed path) so it may be a good idea to provide it.

Reply:
I guess we could, user can still override the `PATH` to test a tpm2-tools patch without relying on an overlay.